src/MDCT.jl

```diff
@@ -63,4 +67,60 @@ end
 imdct(X::AbstractVector{T}) where {T<:Number} =
     imdct(copy!(fftwsimilar(X, size(X)), X))
 
+mutable struct Plan{T<:fftwNumber, N, inv} <: AbstractFFTs.Plan{T}
+    plan::FFTW.r2rFFTWPlan{T} # plan_r2r! REDFT11 plan
+    pinv::AbstractFFTs.Plan{T}
+    Plan{T, N, inv}(plan) where {T, N, inv} = new(plan)
+end
+
+size(p::Plan{T, N, true}) where {T<:Number, N} = (div(N,2), N)
+size(p::Plan{T, N, false}) where {T<:Number, N} = (2*N, N)
+
+function mul!(Y::StridedVector{T}, p::Plan{T, N, false}, X::AbstractVector{T}) where {T<:Number, N}
+    @boundscheck (length(X) == 2*N && length(Y) == N) || throw(DimensionMismatch())
+    N2 = div(N,2)
+    @inbounds for i = 1:N2
+        Y[i] = -0.5 * (X[(3*N2+1)-i] + X[3*N2+i])
+        Y[N2+i] = -0.5 * (X[(2*N2+1)-i] - X[i])
+    end
+    return p.plan * Y
+end
+
+function mul!(Z::StridedVector{T}, p::Plan{T, N, true}, X::AbstractVector{T}) where {T<:Number, N}
+    @boundscheck length(Z) == N || throw(DimensionMismatch())
+    Y = p.plan*copy(X)
```

Comment:
Seems like we should create an out-of-place plan for this case, so that we can just do `Y = p.plan * X`.

Reply:
(i.e. `plan_r2r`, not `plan_r2r!`)

Reply:
https://github.com/stevengj/MDCT.jl/blob/75d4081608dc8ee2fc6610231286e5a67f85824a/src/MDCT.jl#L45
https://github.com/stevengj/MDCT.jl/blob/75d4081608dc8ee2fc6610231286e5a67f85824a/src/MDCT.jl#L63-L64
Now that I look a this I am not sure what to do... Why is `fftwsimilar` needed in the case of `AbstractVector{T}` but not needed when the argument is an `StridedVector{T}`? In both cases `T<:Number`. Also why fftwsimilar doesn't return a `StridedArray{T}`? If this is indeed necessary then maybe it would be better to just change `copy()` to `copy!(fftwsimilar())`, so we don't have to keep both in-place and regular plans in MDC.Plan (although this would decrease performance, because using in-place plan on a copy is less performant then using a regular plan)

Reply:
An `Array` is a subtype of `StridedArray`.